src/components/Signup/Signup.jsx

```diff
@@ -0,0 +1,152 @@
+import { Button, Form, Input, Typography } from 'antd';
+import axios from 'axios';
+import { Link as RouterLink, useNavigate } from 'react-router-dom';
+import { BACKEND_REGISTER_URL } from '../../constants/constants';
+import { useState } from 'react';
+import { validate } from 'email-validator';
+import { message } from 'antd';
+import './Signup.css';
+import { SIGNUP_ALERTS } from '../../constants/constants';
+
+const { Title } = Typography;
+
+const Signup = () => {
+  const navigate = useNavigate();
+  const [signupFormData, setSignupFormData] = useState({
+    name: '',
+    email: '',
+    password: '',
+    confirmPassword: '',
+  });
+
+  const handleInputChange = (fieldName, value) => {
+    setSignupFormData({ ...signupFormData, [fieldName]: value });
+  };
+
+  const isFormDataValidCheck = () => {
+    const { name, email, password, confirmPassword } = signupFormData;
+    const isAllFieldsFilled = name && email && password && confirmPassword;
+    const isEmailValid = validate(email);
+    if (!isAllFieldsFilled) {
+      message.error(SIGNUP_ALERTS.MISSING_FIELDS);
+      return false;
+    }
+    if (password !== confirmPassword) {
+      message.error(SIGNUP_ALERTS.PASSWORDS_DONT_MATCH);
+      return false;
```

Comment:
Not a must but you could extract your repeating code into objects and have simple check instead of all the if conditions